lib/rubygems/source.rb

```diff
@@ -1,4 +1,5 @@
 # frozen_string_literal: true
+autoload :FileUtils, 'fileutils'
```

Comment:
I saw that there has recently been some changes with loading fileutils. I'm not sure what the context is there, so I'm not very confident that this is correct. However, without loading fileutils here, the bundler tests fail and checking for updates does not work. The traceback looks something like:
```
 Invoking `gem list` failed with output:
       ----------------------------------------------------------------------
       /home/runner/work/rubygems/rubygems/lib/rubygems/source.rb:182:in `load_specs': uninitialized constant Gem::Source::FileUtils (NameError)
       	from /home/runner/work/rubygems/rubygems/lib/rubygems/spec_fetcher.rb:256:in `tuples_for'
       	from /home/runner/work/rubygems/rubygems/lib/rubygems/spec_fetcher.rb:220:in `block in available_specs'
       	from /home/runner/work/rubygems/rubygems/lib/rubygems/source_list.rb:98:in `each'
       	from /home/runner/work/rubygems/rubygems/lib/rubygems/source_list.rb:98:in `each_source'
       	from /home/runner/work/rubygems/rubygems/lib/rubygems/spec_fetcher.rb:216:in `available_specs'
       	from /home/runner/work/rubygems/rubygems/lib/rubygems/spec_fetcher.rb:90:in `search_for_dependency'
       	from /home/runner/work/rubygems/rubygems/lib/rubygems/spec_fetcher.rb:153:in `spec_for_dependency'
       	from /home/runner/work/rubygems/rubygems/lib/rubygems.rb:864:in `latest_spec_for'
       	from /home/runner/work/rubygems/rubygems/lib/rubygems.rb:883:in `latest_version_for'
       	from /home/runner/work/rubygems/rubygems/lib/rubygems.rb:875:in `latest_rubygems_version'
       	from /home/runner/work/rubygems/rubygems/lib/rubygems/gem_runner.rb:86:in `warn_on_outdated'
       	from /home/runner/work/rubygems/rubygems/lib/rubygems/gem_runner.rb:59:in `run'
       	from /home/runner/.rubies/ruby-2.4.10/bin/gem:21:in `<main>'
```

Reply:
Yes, the context was that I removed a global fileutils `autoload` like that one in other to workaround some `jruby` code loading bugs (where things apparently sporadically crash if you both `autoload` a constant and also directly `require` the code defining it). So, if the changes in this PR make a missing `FileUtils` constant error surface, I'd rather add the `require` inline right before the error, to keep it consistent and don't hit the same bug in jruby again.

Reply:
ah, I see! Thanks for the context